### PR TITLE
Widget core, new highlight widget class

### PR DESCRIPTION
**Pull request by ppizarror**
As discussed in #156, this introduces a new architecture in widgets:

```
widget/core/
widget/selection/
widget/widgets/
```

Selection class, not yet created, will be fired at `widget.draw_selected_rect()`. This class will accept a rect and return a new surface with the selection.

By default, highlight.py will support the actual behaviour.
Menu accepts as a parameter widget_selector = highlight object

@eforgacs Will create LeftArrow class as a new addition.

### CODE REVIEW

**Comment by anxuae** [comment]
I noticed in your code lots of `_` in front of your variable, what is the purpose?
It is not really [PEP8](https://www.python.org/dev/peps/pep-0008/) compliant (global variables in uppercase, all other in lower snake_case, class attributes start with `_` if really should not be used outside).
IMHO it reduces the readability of the code, but I will do it if it is the convention used for `pygame-menu`, your the boss ;)

**Comment by anxuae** [comment]
I fact regarding the `Selection`, `NoneSelection` and `HighlightSelection` classes, this should be part of the Theme class.

```python
theme = Theme(...)

theme.apply_selected(surface, widget)
```

Moreover it is less painful for end-user to have only one object to create and pass it to the `Menu` class. Also easier to explain and document ;-)

**Comment by anxuae** [comment]
The `set_selection_effect` should be renamed to `set_theme`, this method will be implemented in menu and widget base class.

File: pygameMenu/widgets/selection/highlight.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        Draw the selection.
+
+        :param surface: Surface to draw
+        :type surface: pygame.surface.SurfaceType
+        :param widget: Widget object
+        :type widget: pygameMenu.widgets.core.widget.Widget
+        :return: None
+        """
+        # noinspection PyProtectedMember
+        rect = widget._rect.copy().inflate(self.margin_x, self.margin_y).move(0, -1)
```
Comment (anxuae) [inline pygameMenu/widgets/selection/highlight.py:86]: use `widget.get_rect()`

`widget.get_rect()` should return a copy of the rect by default, it is safer

**Comment by ppizarror** [comment]
> I noticed in your code lots of `_` in front of your variables, what is the purpose?
> 
> It is not really [PEP8](https://www.python.org/dev/peps/pep-0008/) compliant (global variables in uppercase, all other in lower snake_case, class attributes start with `_` if really should not be used outside).
> IMHO it reduces the readability of the code, but I will do it if it is the convention used for `pygame-menu`, your the boss ;)

yep you're right, I don't like it at all... I'm going to fix it

**Comment by ppizarror** [comment]
> I fact regarding the `Selection`, `NoneSelection` and `HighlightSelection` classes, this should be part of the Theme class.
> 
> ```python
> theme = Theme(...)
> 
> theme.apply_selected(surface, widget)
> ```
> 
> if not selection, thus don't apply it
> 
> Moreover it is less painful for end-user to have only one object to create and pass it to the `Menu` class. Also easier to explain and document ;-)

True! I'll call by now theme. Theme also will manage selection_color...

File: pygameMenu/widgets/selection/highlight.py (hunk 1 above)
Comment (ppizarror) [reply pygameMenu/widgets/selection/highlight.py:86]: epic!

**Comment by anxuae** [comment]
I have push a proposal for the Theme class.
All parameters to present in it remains managed by the menu class.

I'm not sure if  font and font_size of element shall be in the theme

**Comment by ppizarror** [comment]
This PR is way too large. I'm gonna merge and still discuss themes in #162

**Comment by eforgacs** [comment]
> This PR is way too large. I'm gonna merge and still discuss themes in #162

Okay, I guess I'll add the LeftArrow class to #162 then

**Pull request merged**